dali/ft/filecopy.cpp

```diff
@@ -465,29 +465,32 @@ FileSizeThread::FileSizeThread(FilePartInfoArray & _queue, CriticalSection & _cs
 
 bool FileSizeThread::wait(unsigned timems)
 {
-    while (!sem.wait(timems))   { // report every time
-        if (!cur.get())
-            continue;       // window here?
-        cur->Link();
+    while (!sem.wait(timems))
+    { // report every time
         RemoteFilename *rfn=NULL;
-        if (copy) {
-            if (!cur->mirrorFilename.isNull())
-                rfn = &cur->mirrorFilename;
-        }
-        else {
-            rfn = &cur->filename;
+        cs.enter();
+        if (cur.get())
+        {
+            if (copy)
+            {
+                if (!cur->mirrorFilename.isNull())
+                    rfn = &cur->mirrorFilename;
+            }
+            else
+            {
+                rfn = &cur->filename;
+            }
         }
-        if (rfn) {
+        cs.leave();
```

Comment:
Should use CriticalBlock b(cs), rather than separate enter(), leave() calls in general, because cleaner and ensures leave() happens.